Pico build on macOS fails because pioasm missing

After install Pico support on macOS Monterey (12.3.1) building helloworld...

```
xs-dev run --example helloworld --device pico
```

...fails because `pioasm` can't be found.

<details>

```
$ xs-dev run --example helloworld --device pico
⠧ Building and deploying project /Users/peterhoddie/.local/share/moddable/examples/helloworld on pico/usr/local/lib/node_modules/xs-dev/node_modules/gluegun/build/index.js:15
    throw up;
    ^

Error: Command failed with exit code 2: mcconfig -m -p pico -t all -o /Users/peterhoddie/.local/share/moddable/build -d
make: /Users/peterhoddie/pico/pico-sdk/build/pioasm/pioasm: No such file or directory
make: *** [/Users/peterhoddie/.local/share/moddable/build/tmp/pico/debug/helloworld/ws2812.pio.h] Error 1
make: *** Waiting for unfinished jobs....
/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c: In function 'fxCreateMachinePlatform':
/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c:79:2: warning: implicit declaration of function 'modMachineTaskInit' [-Wimplicit-function-declaration]
   79 |  modMachineTaskInit(the);
      |  ^~~~~~~~~~~~~~~~~~
/Users/peterhoddie/.local/share/moddable/xs/platforms/mc/xsHosts.c: In function 'modRunMachineSetup':
/Users/peterhoddie/.local/share/moddable/xs/platforms/mc/xsHosts.c:559:2: warning: implicit declaration of function 'modInstrumentationSetup'; did you mean 'modInstrumentationSet_'? [-Wimplicit-function-declaration]
  559 |  modInstrumentationSetup(the);
      |  ^~~~~~~~~~~~~~~~~~~~~~~
      |  modInstrumentationSet_
/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c: In function 'fxDeleteMachinePlatform':
/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c:103:2: warning: implicit declaration of function 'modMachineTaskUninit' [-Wimplicit-function-declaration]
  103 |  modMachineTaskUninit(the);
      |  ^~~~~~~~~~~~~~~~~~~~
/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c: In function 'fxReceiveLoop':
/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c:462:6: warning: implicit declaration of function 'modMessagePostToMachine' [-Wimplicit-function-declaration]
  462 |      modMessagePostToMachine(current, NULL, 0xffff, doDebugCommand, current);
      |      ^~~~~~~~~~~~~~~~~~~~~~~
/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c: In function 'fxInNetworkDebugLoop':
/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c:723:66: warning: comparison between pointer and integer
  723 |  return the->DEBUG_LOOP && the->connection && (kSerialConnection != the->connection);
      |                                                                  ^~
/var/folders/nd/2xq97b3j4fgd9zqrz1hlykvm0000gn/T//ccxgp5iL.s: Assembler messages:
/var/folders/nd/2xq97b3j4fgd9zqrz1hlykvm0000gn/T//ccxgp5iL.s:457: Warning: setting incorrect section attributes for .data.xsram
/var/folders/nd/2xq97b3j4fgd9zqrz1hlykvm0000gn/T//ccC4rD4c.s: Assembler messages:
/var/folders/nd/2xq97b3j4fgd9zqrz1hlykvm0000gn/T//ccC4rD4c.s:2846: Warning: setting incorrect section attributes for .rodata
# compile pio:  ws2812.pio
# library xs: xsHosts.c
# library xs: xsHost.c
# library xs: xsPlatform.c
# library xs: xsAll.c
    at makeError (/usr/local/lib/node_modules/xs-dev/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/usr/local/lib/node_modules/xs-dev/node_modules/execa/index.js:118:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  shortMessage: 'Command failed with exit code 2: mcconfig -m -p pico -t all -o /Users/peterhoddie/.local/share/moddable/build -d',
  command: 'mcconfig -m -p pico -t all -o /Users/peterhoddie/.local/share/moddable/build -d',
  escapedCommand: 'mcconfig -m -p pico -t all -o "/Users/peterhoddie/.local/share/moddable/build" -d',
  exitCode: 2,
  signal: undefined,
  signalDescription: undefined,
  stdout: '# compile pio:  ws2812.pio\n' +
    '# library xs: xsHosts.c\n' +
    '# library xs: xsHost.c\n' +
    '# library xs: xsPlatform.c\n' +
    '# library xs: xsAll.c',
  stderr: 'make: /Users/peterhoddie/pico/pico-sdk/build/pioasm/pioasm: No such file or directory\n' +
    'make: *** [/Users/peterhoddie/.local/share/moddable/build/tmp/pico/debug/helloworld/ws2812.pio.h] Error 1\n' +
    'make: *** Waiting for unfinished jobs....\n' +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c: In function 'fxCreateMachinePlatform':\n" +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c:79:2: warning: implicit declaration of function 'modMachineTaskInit' [-Wimplicit-function-declaration]\n" +
    '   79 |  modMachineTaskInit(the);\n' +
    '      |  ^~~~~~~~~~~~~~~~~~\n' +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/mc/xsHosts.c: In function 'modRunMachineSetup':\n" +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/mc/xsHosts.c:559:2: warning: implicit declaration of function 'modInstrumentationSetup'; did you mean 'modInstrumentationSet_'? [-Wimplicit-function-declaration]\n" +
    '  559 |  modInstrumentationSetup(the);\n' +
    '      |  ^~~~~~~~~~~~~~~~~~~~~~~\n' +
    '      |  modInstrumentationSet_\n' +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c: In function 'fxDeleteMachinePlatform':\n" +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c:103:2: warning: implicit declaration of function 'modMachineTaskUninit' [-Wimplicit-function-declaration]\n" +
    '  103 |  modMachineTaskUninit(the);\n' +
    '      |  ^~~~~~~~~~~~~~~~~~~~\n' +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c: In function 'fxReceiveLoop':\n" +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c:462:6: warning: implicit declaration of function 'modMessagePostToMachine' [-Wimplicit-function-declaration]\n" +
    '  462 |      modMessagePostToMachine(current, NULL, 0xffff, doDebugCommand, current);\n' +
    '      |      ^~~~~~~~~~~~~~~~~~~~~~~\n' +
    "/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c: In function 'fxInNetworkDebugLoop':\n" +
    '/Users/peterhoddie/.local/share/moddable/xs/platforms/pico/xsPlatform.c:723:66: warning: comparison between pointer and integer\n' +
    '  723 |  return the->DEBUG_LOOP && the->connection && (kSerialConnection != the->connection);\n' +
    '      |                                                                  ^~\n' +
    '/var/folders/nd/2xq97b3j4fgd9zqrz1hlykvm0000gn/T//ccxgp5iL.s: Assembler messages:\n' +
    '/var/folders/nd/2xq97b3j4fgd9zqrz1hlykvm0000gn/T//ccxgp5iL.s:457: Warning: setting incorrect section attributes for .data.xsram\n' +
    '/var/folders/nd/2xq97b3j4fgd9zqrz1hlykvm0000gn/T//ccC4rD4c.s: Assembler messages:\n' +
    '/var/folders/nd/2xq97b3j4fgd9zqrz1hlykvm0000gn/T//ccC4rD4c.s:2846: Warning: setting incorrect section attributes for .rodata',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

Node.js v18.12.1
```

</details>

It may be that the Pico setup code in xs-dev needs to be updated. The dependency on `pioasm` is relatively new, arriving when Pico W platform support was implemented. @mkellner maybe able to provide some information to help with this.

--- a/src/extensions/setup-extension.ts
+++ b/src/extensions/setup-extension.ts
@@ -10,8 +10,11 @@ import setupPico from '../toolbox/setup/pico'
 export default async (toolbox: XSDevToolbox): Promise<void> => {
   toolbox.setup = {
     darwin: setupMac,
+    mac: setupMac,
     linux: setupLinux,
+    lin: setupLinux,
     windows_nt: setupWindows,
+    win: setupWindows,
     esp: setupESP8266,
     esp8266: setupESP8266,
     esp32: setupESP32,

--- a/src/extensions/update-extension.ts
+++ b/src/extensions/update-extension.ts
@@ -10,8 +10,11 @@ import updatePico from '../toolbox/update/pico'
 export default async (toolbox: XSDevToolbox): Promise<void> => {
   toolbox.update = {
     darwin: updateMac,
+    mac: updateMac,
     linux: updateLinux,
+    lin: updateLinux,
     windows_nt: updateWindows,
+    win: updateWindows,
     esp: updateESP8266,
     esp8266: updateESP8266,
     esp32: updateESP32,

--- a/src/toolbox/build/index.ts
+++ b/src/toolbox/build/index.ts
@@ -1,7 +1,11 @@
+import { type as platformType } from 'os'
 import handler from 'serve-handler'
 import { createServer } from 'http'
 import { filesystem, print, prompt, system } from 'gluegun'
 import { collectChoicesFromTree } from '../prompt/choices'
+import { moddableExists } from '../setup/moddable'
+import { DEVICE_ALIAS } from '../prompt/devices'
+import { Device } from '../../types'
 
 export type DeployStatus = 'none' | 'run' | 'push'
 
@@ -28,6 +32,8 @@ export async function build({
   deployStatus,
   outputDir,
 }: BuildArgs): Promise<void> {
+  const OS = platformType().toLowerCase() as Device
+
   if (listDevices) {
     const choices = [
       'esp',
@@ -81,6 +87,41 @@ export async function build({
     } else {
       print.warning('Please select a target device to run')
       process.exit(0)
+    }
+  }
+
+  if (targetPlatform !== '') {
+    if (!moddableExists()) {
+      print.error(
+        `Moddable tooling required. Run 'xs-dev setup --device ${DEVICE_ALIAS[OS]}' before trying again.`
+      )
+      process.exit(1)
+    }
+    // preflight / verify environment set up for target platform
+    if (targetPlatform.includes('esp32')) {
+      if (typeof process.env.IDF_PATH !== 'string' || filesystem.exists(process.env.IDF_PATH) !== 'dir') {
+        print.error('The current environment does not appear to be set up for the ESP32 build target. Please run `xs-dev setup --device esp32` before trying again.')
+        process.exit(1)
+      }
+    } else if (targetPlatform.includes('esp')) {
+      if (typeof process.env.ESP_BASE !== 'string' || filesystem.exists(process.env.ESP_BASE) !== 'dir') {
+        print.error('The current environment does not appear to be set up for the ESP8266 build target. Please run `xs-dev setup --device esp8266` before trying again.')
+        process.exit(1)
+      }
+    }
+
+    if (targetPlatform.includes('wasm')) {
+      if (!(process.env.PATH ?? '').includes('binaryen') || filesystem.exists(process.env.EMSDK ?? '') !== 'dir' || filesystem.exists(process.env.EMSDK_NODE ?? '') !== 'file' || filesystem.exists(process.env.EMSDK_PYTHON ?? '') !== 'file') {
+        print.error('The current environment does not appear to be set up for the wasm build target. Please run `xs-dev setup --device wasm` before trying again.')
+        process.exit(1)
+      }
+    }
+
+    if (targetPlatform.includes('pico')) {
+      if (typeof process.env.PICO_SDK_PATH !== 'string' || filesystem.exists(process.env.PICO_SDK_PATH) !== 'dir' || system.which('picotool') === null || filesystem.exists(process.env.PIOASM ?? '') !== 'file') {
+        print.error('The current environment does not appear to be set up for the pico build target. Please run `xs-dev setup --device pico` before trying again.')
+        process.exit(1)
+      }
     }
   }
 
@@ -142,8 +183,7 @@ export async function build({
   const spinner = print.spin()
 
   spinner.start(
-    `Building${
-      deployStatus !== 'none' ? ' and deploying project' : ''
+    `Building${deployStatus !== 'none' ? ' and deploying project' : ''
     } ${projectPath} on ${targetPlatform}`
   )
 

--- a/src/toolbox/build/index.ts
+++ b/src/toolbox/build/index.ts
@@ -34,6 +34,13 @@ export async function build({
 }: BuildArgs): Promise<void> {
   const OS = platformType().toLowerCase() as Device
 
+  if (!moddableExists()) {
+    print.error(
+      `Moddable tooling required. Run 'xs-dev setup --device ${DEVICE_ALIAS[OS]}' before trying again.`
+    )
+    process.exit(1)
+  }
+
   if (listDevices) {
     const choices = [
       'esp',
@@ -91,13 +98,6 @@ export async function build({
   }
 
   if (targetPlatform !== '') {
-    if (!moddableExists()) {
-      print.error(
-        `Moddable tooling required. Run 'xs-dev setup --device ${DEVICE_ALIAS[OS]}' before trying again.`
-      )
-      process.exit(1)
-    }
-    // preflight / verify environment set up for target platform
     if (targetPlatform.includes('esp32')) {
       if (typeof process.env.IDF_PATH !== 'string' || filesystem.exists(process.env.IDF_PATH) !== 'dir') {
         print.error('The current environment does not appear to be set up for the ESP32 build target. Please run `xs-dev setup --device esp32` before trying again.')

--- a/src/toolbox/prompt/devices.ts
+++ b/src/toolbox/prompt/devices.ts
@@ -3,8 +3,11 @@ import { Device } from '../../types'
 export const DEVICE_ALIAS: Record<Device | 'esp8266', string> = Object.freeze({
   esp8266: 'esp',
   darwin: 'mac',
+  mac: 'mac',
   windows_nt: 'win',
+  win: 'win',
   linux: 'lin',
+  lin: 'lin',
   esp: 'esp',
   esp32: 'esp32',
   wasm: 'wasm',

--- a/src/toolbox/setup/esp32.ts
+++ b/src/toolbox/setup/esp32.ts
@@ -8,7 +8,7 @@ import { installDeps as installLinuxDeps } from './esp32/linux'
 import { installDeps as installWinDeps } from './esp32/windows'
 import { setEnv, ensureModdableCommandPrompt } from './windows'
 
-export default async function (): Promise<void> {
+export default async function(): Promise<void> {
   const OS = platformType().toLowerCase()
   const isWindows = OS === "windows_nt"
   const ESP_IDF_REPO = 'https://github.com/espressif/esp-idf.git'
@@ -27,8 +27,9 @@ export default async function (): Promise<void> {
     process.exit(1)
   }
 
-  if (isWindows)
+  if (isWindows) {
     await ensureModdableCommandPrompt(spinner)
+  }
 
   // 1. ensure ~/.local/share/esp32 directory
   spinner.info('Ensuring esp32 install directory')
@@ -102,6 +103,6 @@ export default async function (): Promise<void> {
   spinner.succeed(`
   Successfully set up esp32 platform support for Moddable!
   Test out the setup by starting a new ${isWindows ? 'Moddable Command Prompt' : 'terminal session'}, plugging in your device, and running: xs-dev run --example helloworld --device=esp32
-  If there is trouble finding the correct port, pass the "--port" flag to the above command with the ${isWindows? "COM Port" :  "path to the /dev.cu.*"} that matches your device.
+  If there is trouble finding the correct port, pass the "--port" flag to the above command with the ${isWindows ? "COM Port" : "path to the /dev.cu.*"} that matches your device.
   `)
 }

--- a/src/toolbox/setup/esp8266.ts
+++ b/src/toolbox/setup/esp8266.ts
@@ -13,11 +13,13 @@ import { installDeps as installMacDeps } from './esp8266/mac'
 import { installDeps as installLinuxDeps } from './esp8266/linux'
 import { installDeps as installWindowsDeps } from './esp8266/windows'
 import { ensureModdableCommandPrompt } from './windows'
+import { DEVICE_ALIAS } from '../prompt/devices'
+import { Device } from '../../types'
 
 const finishedPromise = promisify(finished)
 
-export default async function (): Promise<void> {
-  const OS = platformType().toLowerCase()
+export default async function(): Promise<void> {
+  const OS = platformType().toLowerCase() as Device
   const isWindows = OS === "windows_nt"
   const TOOLCHAIN = isWindows ? 'https://www.dropbox.com/s/edbtq6oi1up623i/esp8266.toolchain.win32.zip?dl=1' : `https://www.moddable.com/private/esp8266.toolchain.${OS}.tgz`
   const ARDUINO_CORE =
@@ -35,24 +37,25 @@ export default async function (): Promise<void> {
   // 0. ensure Moddable exists
   if (!moddableExists()) {
     spinner.fail(
-      'Moddable tooling required. Run `xs-dev setup` before trying again.'
+      `Moddable tooling required. Run 'xs-dev setup --device ${DEVICE_ALIAS[OS]}' before trying again.`
     )
     process.exit(1)
   }
 
-  if (isWindows)
+  if (isWindows) {
     await ensureModdableCommandPrompt(spinner)
+  }
 
   // 1. ensure ~/.local/share/esp directory
-  spinner.info('Ensuring esp directory')
+  spinner.info('Ensuring esp8266 directory')
   filesystem.dir(ESP_DIR)
 
   // 2. download and untar xtensa toolchain
   if (filesystem.exists(TOOLCHAIN_PATH) === false) {
     spinner.start('Downloading xtensa toolchain')
-    
+
     if (isWindows) {
-      const writer = ZipExtract({path: ESP_DIR})
+      const writer = ZipExtract({ path: ESP_DIR })
       const response = await axios.get(TOOLCHAIN, {
         responseType: 'stream'
       })

--- a/src/toolbox/setup/esp8266/mac.ts
+++ b/src/toolbox/setup/esp8266/mac.ts
@@ -6,8 +6,15 @@ export async function installDeps(
 ): Promise<void> {
   if (system.which('python') === null) {
     spinner.start('Installing python from homebrew')
-    await system.exec('brew install python')
-    spinner.succeed()
+    try {
+      await system.exec('brew install python')
+      spinner.succeed()
+    } catch (error: unknown) {
+      if (error instanceof Error && error.toString().includes('xcode-select')) {
+        spinner.fail('Apple Command Line Tools must be installed in order to install python from Homebrew. Please run `xcode-select --install` before trying again.')
+        process.exit(1)
+      }
+    }
   }
 
   if (system.which('pip') === null) {

--- a/src/toolbox/setup/esp8266/mac.ts
+++ b/src/toolbox/setup/esp8266/mac.ts
@@ -10,7 +10,7 @@ export async function installDeps(
       await system.exec('brew install python')
       spinner.succeed()
     } catch (error: unknown) {
-      if (error instanceof Error && error.toString().includes('xcode-select')) {
+      if (error instanceof Error && error.message.includes('xcode-select')) {
         spinner.fail('Apple Command Line Tools must be installed in order to install python from Homebrew. Please run `xcode-select --install` before trying again.')
         process.exit(1)
       }

--- a/src/toolbox/setup/esp8266/windows.ts
+++ b/src/toolbox/setup/esp8266/windows.ts
@@ -14,74 +14,74 @@ const ESP_TOOL_VERSION = 'esptool-0.4.13-win32'
 const CYGWIN = 'https://www.dropbox.com/s/ub7xehxbf747eu1/cygwin.win32.zip?dl=1'
 
 export async function installPython(spinner: ReturnType<GluegunPrint['spin']>): Promise<void> {
-    if (system.which('python') === null) {
-        // For some reason, system.which does not work with winget. This is a workaround for now.
-        try {
-            await system.exec('where winget')
-        } catch (error) {
-            print.error('Python is required.')
-            print.info('You can download and install Python from python.org/downloads')
-            print.info('Or xs-dev can manage installing Python and other dependencies using the Windows Package Manager Client (winget).')
-            print.info('You can install winget via the App Installer package in the Microsoft Store.')
-            print.info('Please install either Python or winget, then launch a new Command Prompt and re-run this setup.')
-            throw new Error("Python is required")
-        }
-        
-        spinner.start('Installing python from winget')
-        await system.exec('winget install -e --id Python.Python.3 --silent')
-        spinner.succeed()
-        print.info('Python successfully installed. Please close this window and launch a new Moddable Command Prompt to refresh environment variables, then re-run this setup.')
-        throw new Error("Command Prompt restart needed")
+  if (system.which('python') === null) {
+    // For some reason, system.which does not work with winget. This is a workaround for now.
+    try {
+      await system.exec('where winget')
+    } catch (error) {
+      print.error('Python is required.')
+      print.info('You can download and install Python from python.org/downloads')
+      print.info('Or xs-dev can manage installing Python and other dependencies using the Windows Package Manager Client (winget).')
+      print.info('You can install winget via the App Installer package in the Microsoft Store.')
+      print.info('Please install either Python or winget, then launch a new Command Prompt and re-run this setup.')
+      throw new Error("Python is required")
     }
+
+    spinner.start('Installing python from winget')
+    await system.exec('winget install -e --id Python.Python.3 --silent')
+    spinner.succeed()
+    print.info('Python successfully installed. Please close this window and launch a new Moddable Command Prompt to refresh environment variables, then re-run this setup.')
+    throw new Error("Command Prompt restart needed")
+  }
 }
 
 export async function installDeps(
   spinner: ReturnType<GluegunPrint['spin']>,
   ESP_DIR: string
 ): Promise<void> {
-    const ESP_TOOL_DIR = filesystem.resolve(ESP_DIR, ESP_TOOL_VERSION)
-    const ESP_TOOL_EXE = filesystem.resolve(ESP_TOOL_DIR, 'esptool.exe')
-    const ESP_TOOL_DESTINATION = filesystem.resolve(ESP_DIR, 'esptool.exe')
-    const CYGWIN_BIN = filesystem.resolve(ESP_DIR, "cygwin", "bin")
+  const ESP_TOOL_DIR = filesystem.resolve(ESP_DIR, ESP_TOOL_VERSION)
+  const ESP_TOOL_EXE = filesystem.resolve(ESP_TOOL_DIR, 'esptool.exe')
+  const ESP_TOOL_DESTINATION = filesystem.resolve(ESP_DIR, 'esptool.exe')
+  const CYGWIN_BIN = filesystem.resolve(ESP_DIR, "cygwin", "bin")
 
-    spinner.start('Downloading ESP Tool')
-    let writer = ZipExtract({path: ESP_DIR})
-    let response = await axios.get(ESP_TOOL, {
-        responseType: 'stream'
-    })
-    response.data.pipe(writer)
-    await finishedPromise(writer)
-    filesystem.move(ESP_TOOL_EXE, ESP_TOOL_DESTINATION, {overwrite: true})
-    filesystem.remove(ESP_TOOL_DIR)
-    spinner.succeed()
+  spinner.start('Downloading ESP Tool')
+  let writer = ZipExtract({ path: ESP_DIR })
+  let response = await axios.get(ESP_TOOL, {
+    responseType: 'stream'
+  })
+  response.data.pipe(writer)
+  await finishedPromise(writer)
+  filesystem.move(ESP_TOOL_EXE, ESP_TOOL_DESTINATION, { overwrite: true })
+  filesystem.remove(ESP_TOOL_DIR)
+  spinner.succeed()
 
-    spinner.start('Downloading Cygwin toolchain support package')
-    writer = ZipExtract({path: ESP_DIR})
-    response = await axios.get(CYGWIN, {
-        responseType: 'stream'
-    })
-    response.data.pipe(writer)
-    await finishedPromise(writer)
-    spinner.succeed()
+  spinner.start('Downloading Cygwin toolchain support package')
+  writer = ZipExtract({ path: ESP_DIR })
+  response = await axios.get(CYGWIN, {
+    responseType: 'stream'
+  })
+  response.data.pipe(writer)
+  await finishedPromise(writer)
+  spinner.succeed()
 
-    spinner.start('Setting environment variables')
-    await setEnv("BASE_DIR", INSTALL_DIR)
-    await addToPath(CYGWIN_BIN)
-    spinner.succeed()
+  spinner.start('Setting environment variables')
+  await setEnv("BASE_DIR", INSTALL_DIR)
+  await addToPath(CYGWIN_BIN)
+  spinner.succeed()
 
-    try {
-        await installPython(spinner)
-    } catch (error) { // Command Prompt restart needed
-        process.exit(1)
-    }
-    
-    if (system.which('pip') === null) {
-        spinner.start('Installing pip through ensurepip')
-        await system.exec('python -m ensurepip')
-        spinner.succeed()
-    }
-    
-    spinner.start('Installing pyserial through pip')
-    await system.exec('python -m pip install pyserial')
+  try {
+    await installPython(spinner)
+  } catch (error) { // Command Prompt restart needed
+    process.exit(1)
+  }
+
+  if (system.which('pip') === null) {
+    spinner.start('Installing pip through ensurepip')
+    await system.exec('python -m ensurepip')
     spinner.succeed()
+  }
+
+  spinner.start('Installing pyserial through pip')
+  await system.exec('python -m pip install pyserial')
+  spinner.succeed()
 }

--- a/src/toolbox/setup/moddable.ts
+++ b/src/toolbox/setup/moddable.ts
@@ -1,16 +1,42 @@
+import { type as platformType } from 'os'
 import { finished } from 'stream'
 import { promisify } from 'util'
 import { filesystem } from 'gluegun'
 import { Octokit, RestEndpointMethodTypes } from '@octokit/rest'
 import { Extract as ZipExtract } from 'unzip-stream'
 import axios from 'axios'
+import { INSTALL_PATH } from './constants'
+import { Device } from '../../types'
+import { DEVICE_ALIAS } from '../prompt/devices'
 
 const finishedPromise = promisify(finished)
 
 export function moddableExists(): boolean {
+  const OS = platformType().toLowerCase() as Device
+  const platformDir = DEVICE_ALIAS[OS]
+  const releaseTools = filesystem.exists(
+    filesystem.resolve(
+      INSTALL_PATH,
+      'build',
+      'bin',
+      platformDir,
+      'release'
+    )
+  )
+  const debugTools = filesystem.exists(
+    filesystem.resolve(
+      INSTALL_PATH,
+      'build',
+      'bin',
+      platformDir,
+      'debug'
+    )
+  )
   return (
     process.env.MODDABLE !== undefined &&
-    filesystem.exists(process.env.MODDABLE) === 'dir'
+    filesystem.exists(process.env.MODDABLE) === 'dir' &&
+    releaseTools === 'dir' &&
+    debugTools === 'dir'
   )
 }
 

--- a/src/toolbox/setup/pico.ts
+++ b/src/toolbox/setup/pico.ts
@@ -4,34 +4,36 @@ import { INSTALL_DIR, EXPORTS_FILE_PATH } from './constants'
 import upsert from '../patching/upsert'
 import { installDeps as installMacDeps } from './pico/mac'
 import { installDeps as installLinuxDeps } from './pico/linux'
+import { moddableExists } from './moddable'
 
-export default async function (): Promise<void> {
+export default async function(): Promise<void> {
   const OS = platformType().toLowerCase()
   const PICO_SDK_REPO = 'https://github.com/raspberrypi/pico-sdk'
   const PICO_EXAMPLES_REPO = 'https://github.com/raspberrypi/pico-examples'
   const PICOTOOL_REPO = 'https://github.com/raspberrypi/picotool'
-  const PICO_DIR = filesystem.resolve(INSTALL_DIR, 'pico')
-  const PICO_SDK_DIR = filesystem.resolve(PICO_DIR, 'pico-sdk')
-  const PICO_EXAMPLES_PATH = filesystem.resolve(PICO_DIR, 'pico-examples')
-  const PICOTOOL_PATH = filesystem.resolve(PICO_DIR, 'picotool')
+  const PICO_ROOT = process.env.PICO_ROOT ?? filesystem.resolve(INSTALL_DIR, 'pico')
+  const PICO_SDK_DIR = filesystem.resolve(PICO_ROOT, 'pico-sdk')
+  const PICO_EXAMPLES_PATH = filesystem.resolve(PICO_ROOT, 'pico-examples')
+  const PICOTOOL_PATH = filesystem.resolve(PICO_ROOT, 'picotool')
   const PICOTOOL_BUILD_DIR = filesystem.resolve(PICOTOOL_PATH, 'build')
   const PICO_SDK_BUILD_DIR = filesystem.resolve(PICO_SDK_DIR, 'build')
+  const PIOASM_PATH = filesystem.resolve(PICO_SDK_BUILD_DIR, 'pioasm', 'pioasm')
 
   const spinner = print.spin()
   spinner.start('Starting pico tooling setup')
 
   // 0. ensure pico instal directory and Moddable exists
-  if (
-    process.env.MODDABLE === undefined ||
-    filesystem.exists(process.env.MODDABLE) === false
-  ) {
+  if (!moddableExists()) {
     spinner.fail(
       'Moddable platform tooling required. Run `xs-dev setup` before trying again.'
     )
     process.exit(1)
   }
   spinner.info('Ensuring pico directory')
-  filesystem.dir(PICO_DIR)
+  filesystem.dir(PICO_ROOT)
+  if (process.env.PICO_ROOT === undefined) {
+    await upsert(EXPORTS_FILE_PATH, `export PICO_ROOT=${PICO_ROOT}`)
+  }
 
   // 1. Install required components
   if (OS === 'darwin') {
@@ -125,6 +127,14 @@ export default async function (): Promise<void> {
     spinner.info(`Using existing $PICO_SDK_PATH: ${process.env.PICO_SDK_PATH}`)
   }
 
+  if (process.env.PIOASM === undefined) {
+    spinner.info('Setting PIOASM')
+    process.env.PIOASM = PIOASM_PATH
+    await upsert(EXPORTS_FILE_PATH, `export PIOASM=${PIOASM_PATH}`)
+  } else {
+    spinner.info(`Using existing $PIOASM: ${process.env.PIOASM}`)
+  }
+
   if (
     filesystem.exists(PICOTOOL_BUILD_DIR) === false ||
     filesystem.list(PICOTOOL_BUILD_DIR)?.length === 0
@@ -144,6 +154,7 @@ export default async function (): Promise<void> {
     await upsert(EXPORTS_FILE_PATH, `export PATH="${PICOTOOL_BUILD_DIR}:$PATH"`)
     spinner.succeed()
   }
+
 
   spinner.succeed(`
 Successfully set up pico platform support for Moddable!

--- a/src/toolbox/setup/wasm.ts
+++ b/src/toolbox/setup/wasm.ts
@@ -5,7 +5,7 @@ import { moddableExists } from './moddable'
 import upsert from '../patching/upsert'
 import { execWithSudo } from '../system/exec'
 
-export default async function (): Promise<void> {
+export default async function(): Promise<void> {
   const OS = platformType().toLowerCase()
   const EMSDK_REPO = 'https://github.com/emscripten-core/emsdk.git'
   const BINARYEN_REPO = 'https://github.com/WebAssembly/binaryen.git'
@@ -42,22 +42,29 @@ export default async function (): Promise<void> {
   if (shouldBuildEmsdk) {
     try {
       // clear residual env settings
-      process.env.EMSDK = undefined
-      process.env.EMSDK_NODE = undefined
-      process.env.EMSDK_PYTHON = undefined
+      process.env.EMSDK = ""
+      process.env.EMSDK_NODE = ""
+      process.env.EMSDK_PYTHON = ""
 
       spinner.start('Installing latest EMSDK')
+      print.debug(EMSDK_PATH)
       await system.exec('./emsdk install latest', {
+        process: process,
         cwd: EMSDK_PATH,
         stdout: process.stdout,
       })
       await system.exec('./emsdk activate latest', {
+        process: process,
         cwd: EMSDK_PATH,
         stdout: process.stdout,
       })
       await upsert(
         EXPORTS_FILE_PATH,
-        `source ${filesystem.resolve(EMSDK_PATH, 'emsdk_env.sh')}`
+        `export EMSDK_QUIET=1`
+      )
+      await upsert(
+        EXPORTS_FILE_PATH,
+        `source ${filesystem.resolve(EMSDK_PATH, 'emsdk_env.sh')} 1> /dev/null`
       )
     } catch (error) {
       spinner.fail(`Error activating emsdk: ${String(error)}`)

--- a/src/types.ts
+++ b/src/types.ts
@@ -4,8 +4,11 @@ import type { BuildArgs } from './toolbox/build/index'
 
 export type Device =
   | 'darwin'
+  | 'mac'
   | 'linux'
+  | 'lin'
   | 'windows_nt'
+  | 'win'
   | 'esp'
   | 'esp8266'
   | 'esp32'

